rust/core-lib/src/plugins/rpc_types.rs

```diff
@@ -127,10 +127,35 @@ pub struct Span {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+pub struct ScopeSpan {
+    pub start: usize,
+    pub end: usize,
+    pub scope_id: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Style {
+    pub fg: u32,
+    #[serde(rename = "font")]
+    pub font_style: u8,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct StyleDef {
+    pub style: Style,
+    pub scope: String,
```

Comment:
This isn't used yet right?

But when it is used, this should be more like the struct in syntect in that `scope` should be named `selector` since it can be a fancier pattern than a single scope. Also `Style` should probably be a `StyleModifier` like syntect where each of foreground, background, font modifier is optional so they can be merged properly.

You could fix this in this PR but if it is actually not used, I would remove it from this PR and stash it in a branch to work off of later when it actually is used.